llmfoundry/models/layers/llama_attention_monkeypatch.py

```diff
@@ -123,9 +122,11 @@ def llama_attention_patch_torch(
     if past_key_value is not None:
         kv_seq_len += past_key_value[0].shape[-2]
     cos, sin = self.rotary_emb(value_states, seq_len=kv_seq_len)
-    query_states, key_states = apply_rotary_pos_emb(
-        query_states, key_states, cos, sin,
-        position_ids)  # type: ignore (thirdParty)
+
+    if position_ids is None:
```

Comment:
I think `position_ids` actually can be `None` and I typed `apply_rotary_pos_emb` incorrectly when I added types to the huggingface code.

e.g.
```
In [7]: d = torch.randn(2048, 768)

In [8]: d[None]
Out[8]: 
tensor([[[-0.3377, -0.0054,  0.1669,  ..., -0.9896, -0.1753,  0.4629],
         [ 0.6908, -0.6488,  1.0811,  ..., -0.9895, -1.0824,  2.3419],
         [-1.7552, -0.3778,  1.7064,  ..., -0.0380,  0.7781,  0.1165],
         ...,
         [-0.4624, -0.2032,  0.9854,  ..., -0.6765, -1.9464, -0.7511],
         [ 0.6446, -2.3273,  0.0222,  ..., -0.4254, -0.5582, -0.6867],
         [ 2.4439,  0.6429, -0.3800,  ...,  0.1195,  0.0516,  0.7610]]])
```

not entirely sure if the [huggingface code](https://github.com/huggingface/transformers/blob/5af2c6269672cda01c24ad48fab13f14a3ffb746/src/transformers/models/llama/modeling_llama.py#L180-L188) silently assumes non None since it isn't typed.

Reply:
OK I changed the type on `apply_rotary_pos_emb`